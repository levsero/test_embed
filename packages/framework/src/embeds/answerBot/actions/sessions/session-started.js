import { SESSION_STARTED } from './action-types'

const generateSessionID = () => Date.now()

const generateSessionData = () => ({
  resolved: false,
  fallbackSuggested: false,
  requestStatus: null,
  articles: [],
})

export const sessionStarted = () => {
  return {
    type: SESSION_STARTED,
    payload: {
      sessionID: generateSessionID(),
      sessionData: generateSessionData(),
    },
  }
}
