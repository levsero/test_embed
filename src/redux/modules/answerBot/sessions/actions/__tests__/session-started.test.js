import * as actions from '../session-started';

test('sessionStarted generates the expected payload', () => {
  jest.spyOn(Date, 'now').mockReturnValue(22222);
  expect(actions.sessionStarted())
    .toMatchSnapshot();
});
