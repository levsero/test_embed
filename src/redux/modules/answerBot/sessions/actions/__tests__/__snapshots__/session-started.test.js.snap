// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sessionStarted generates the expected payload 1`] = `
Object {
  "payload": Object {
    "sessionData": Object {
      "articles": Array [],
      "fallbackSuggested": false,
      "requestStatus": null,
      "resolved": false,
    },
    "sessionID": 22222,
  },
  "type": "widget/answerBot/SESSION_STARTED",
}
`;
